### ISSUE TITLE
dartpad depends on route_hierarchical, a discontinued package

### ISSUE THREAD

**Issue by jcollins-g** [dart-lang/dart-pad#832]
https://pub.dartlang.org/packages/route_hierarchical

and

https://github.com/dart-lang/dart-pad/blob/44a0320c64bad30b79e247ef947c0a6b8817f642/pubspec.yaml#L14

**Comment by devoncarew** [comment]
We could investigate using another router; `package:m4d_router` looks like it would do what we need.

We could also roll out own - we don't use much from the router. Here's our API exposure:

```dart
class Router {
  final Route root = new Route();

  Future<bool> go(String routePath, Map parameters, {bool forceReload: false}) {
  }

  void listen() {
  }
}

class RouteEnterEvent {
  final Map<String, String> parameters = {};
}

typedef void RouteEnterEventHandler(RouteEnterEvent event);

class Route {
  void addRoute({
    String name,
    Pattern path,
    bool defaultRoute: false,
    RouteEnterEventHandler enter,
  }) {

  }
}

```

**Comment by jcollins-g** [comment]
I've managed to get it working with a modified version of the route_hierarchical package so that's an option as well (to take the whole old package in wholesale).  I'll look into ripping out just the parts we use, too.

**Comment by devoncarew** [comment]
> I've managed to get it working with a modified version of the route_hierarchical package so that's an option as well (to take the whole old package in wholesale). I'll look into ripping out just the parts we use, too.

Awesome!

@MikeMitterer, I see you just recently published `m4d_router`; what are your thoughts for using it in DartPad? We have fairly simple routing needs - we'd mostly want something low maintenance that was an easy replacement for route_hierarchical.

**Comment by jcollins-g** [comment]
Fixed in #834.
